File: app/src/main/java/com/paradigmadigital/dagger/ui/detail/DetailActivity.java
package com.paradigmadigital.dagger.ui.detail;

import com.paradigmadigital.dagger.R;
import com.paradigmadigital.dagger.ui.BaseActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import javax.inject.Inject;

public class DetailActivity extends BaseActivity {

    @Inject
    DetailPresenter presenter;

    View.OnClickListener listener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            presenter.onClick();
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        getActivityComponent().inject(this);

        presenter.initialize();
        ((Button) findViewById(R.id.button)).setOnClickListener(listener);
    }
}

File: module_detail/src/main/java/eu/sesma/dagger/detail/DetailActivity.kt
package eu.sesma.dagger.detail

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import eu.sesma.dagger.R
import org.koin.android.scope.currentScope

class DetailActivity : AppCompatActivity() {

    val presenter: DetailPresenter by currentScope.inject()

    private val listener: View.OnClickListener = View.OnClickListener { presenter.onClick() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        presenter.initialize()
        findViewById<Button>(R.id.button).setOnClickListener(listener)
    }
}
